File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for char in s:
            if char == '(' or char == '{' or char == '[':
                stack.append(char)
            elif char == ')':
                if not stack or stack.pop() != '(':
                    return False
            elif char == '}':
                if not stack or stack.pop() != '{':
                    return False
            elif char == ']':
                if not stack or stack.pop() != '[':
                    return False

        return False if stack else True

File: Data Structures & Algorithms/daily-temperatures/submission-2.cpp
class Solution {
public:
  vector<int> dailyTemperatures(vector<int> &temperatures) {
    stack<pair<int, int>> st;
    vector<int> answer(temperatures.size());
    for (int i = 0; i < temperatures.size(); i++) {
      if (st.empty()) {
        st.push({temperatures[i], i});
      } else {
        while (!st.empty() && st.top().first < temperatures[i]) {
          answer[st.top().second] = i - st.top().second;
          st.pop();
        }
        st.push({temperatures[i], i});
      }
    }
    return answer;
  }
};